components/datetime/src/any/zoned_datetime.rs

```diff
@@ -196,6 +201,38 @@ impl ZonedAnyDateTimeFormatter {
     /// Furthermore, based on the type of calendar used, one of the following data keys may be necessary:
     ///
     /// - `u-ca-japanese` (Japanese calendar): `calendar/japanese@1`
+    ///
+    /// Test will currently fail due to <https://github.com/unicode-org/icu4x/issues/2188>,
+    /// since these functions currently *must* be given a fallback-enabled provider and
+    /// we do not have one in `icu_testdata`
+    ///
+    /// ```rust,should_panic
+    /// use icu::calendar::Gregorian;
+    /// use icu::datetime::options::length;
+    /// use icu::datetime::mock::parse_zoned_gregorian_from_str;
+    /// use icu::datetime::{DateTimeFormatterOptions, any::ZonedAnyDateTimeFormatter};
+    /// use icu::locid::Locale;
+    /// use icu::datetime::TimeZoneFormatterOptions;
+    /// use std::str::FromStr;
+    ///
+    /// let provider = icu_testdata::get_baked_provider();
```

Comment:
Does `icu_testdata::get_provider().as_any_provider()` not work?

Reply:
Nope, StaticDP isn't an AnyP